src/api/app/models/project.rb

```diff
@@ -1404,13 +1404,10 @@ def bsrequest_repos_map(project)
 
   def self.valid_name?(name)
```

Comment:
Where do we use ``valid_name?``? Only as validation? Maybe we can try to move the functionality in plain ActiveModel validations like ``format_of`` and ``length`` etc?